merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/PostgresResultsCellRepository.java

```diff
@@ -449,8 +408,77 @@ public PostgresResultsCell(
     @Override
     public State get() {
       try (final var connection = dataSource.getConnection()) {
-        return getSimulationState(connection, datasetId)
-            .orElseThrow(() -> new Error("Dataset corrupted"));
+        Optional<State> result;
+        final var record$ = getSimulationDatasetRecord(
+            connection,
+            datasetId,
+            planStart);
+        if (record$.isEmpty()) {
+          result = Optional.empty();
+        } else {
+          final var record = record$.get();
+          result = Optional.of(
+              switch (record.state().status()) {
+                case PENDING -> new State.Pending(record.simulationDatasetId());
+                case INCOMPLETE -> new State.Incomplete(record.simulationDatasetId());
+                case FAILED -> new State.Failed(record.simulationDatasetId(), record.state().reason()
+                                                                                    .orElseThrow(() -> new Error(
+                                                                                        "Unexpected state: %s request state has no failure message".formatted(
+                                                                                            record.state().status()))));
+                case SUCCESS -> new State.Success(
+                    record.simulationDatasetId(),
+                    new SimulationResultsHandle() {
+                      @Override
+                      public SimulationResults getSimulationResults() {
```

Comment:
Should this inline class instantiation be extracted to a `PostgresSimulationResultsHandle`?